src/protocol.rs

```diff
@@ -616,27 +593,31 @@ mod tests {
             &message.message_type(),
         );
         let (sid, output, messages) = participant.process_single_message(&message, rng)?;
+        if sid != participant.sid {
+            // The returned Session ID should _always_ be the same as the participant's Session ID.
+            return Err(InternalError::InternalInvariantFailed);
+        }
         deliver_all(&messages, inboxes)?;
 
         // Check the protocol outputs are valid
         assert_eq!(message.id(), sid);
         let is_done_computes_correctly = match output {
-            Output::AuxInfo(_, _) => participant.is_auxinfo_done(sid),
-            Output::KeyGen(_, _) => participant.is_keygen_done(sid),
-            Output::Presign(_) => participant.is_presigning_done(sid),
+            Output::AuxInfo(_, _) => participant.is_auxinfo_done(),
+            Output::KeyGen(_, _) => participant.is_keygen_done(),
+            Output::Presign(_) => participant.is_presigning_done(),
             Output::Sign(_) => Ok(true), // this doesn't have a check
             Output::None => {
-                let auxinfo = participant.is_auxinfo_done(sid);
-                let keygen = participant.is_keygen_done(sid);
-                let presign = participant.is_presigning_done(sid);
+                let auxinfo = participant.is_auxinfo_done();
+                let keygen = participant.is_keygen_done();
+                let presign = participant.is_presigning_done();
 
                 // The current behavior of these is weird -- they return Ok even if the `sid`
                 // corresponds to a different protocol. Perhaps the "most
                 // correct" version of this would check that exactly
                 // one of these returns Ok(false), and the others return Err, but here we are:
-                assert_eq!(auxinfo, Ok(false));
-                assert_eq!(keygen, Ok(false));
-                assert_eq!(presign, Ok(false));
+                // assert_eq!(auxinfo, Ok(false));
```

Comment:
I _think_ this check is no longer relevant, because we can indeed have some protocols be completed despite receiving `Output::None`. For example, `auxinfo` may complete and we could be processing an output associated with `keygen`.